editors/vscode/src/extension.ts

```diff
@@ -170,29 +170,16 @@ async function languageActivate(context: IContext) {
       // if (langId === "plaintext") {
       //     console.log("plaintext", langId, editor?.document.uri.fsPath);
       // }
-      if (!isTypstDocument(editor?.document)) {
-        // console.log("not typst", langId, editor?.document.uri.fsPath);
-        return commandActivateDoc(undefined);
-      }
-      return commandActivateDoc(editor?.document);
+      return focusDoc(isTypstDocument(editor?.document) ? editor?.document : undefined, editor);
     }),
-  );
-  context.subscriptions.push(
     vscode.workspace.onDidOpenTextDocument((doc: vscode.TextDocument) => {
       if (doc.isUntitled && window.activeTextEditor?.document === doc) {
```

Comment:
I think the doc.isUntitled is a mistake, you probably meant !doc.isUntitled, because, just before on line 166 you return if the document is untitled which makes sense because you don't want to consider it.
```suggestion
      if (!doc.isUntitled && window.activeTextEditor?.document === doc) {
```

Reply:
Hello, I added comments to the code. I checked it and think of I'm not doing a bad refactor. Please check the comments:

https://github.com/Myriad-Dreamin/tinymist/blob/f4614eee70efd42b7d2fe133751d8daa5abc6122/editors/vscode/src/extension.ts#L208-L211

https://github.com/Myriad-Dreamin/tinymist/blob/f4614eee70efd42b7d2fe133751d8daa5abc6122/editors/vscode/src/extension.ts#L216-L223